src/certificate.rs

```diff
@@ -31,15 +31,102 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 use crate::{
-    consts::*, error::Error, key::SlotId, serialization::*, transaction::Transaction,
-    yubikey::YubiKey, Buffer,
+    consts::*,
+    error::Error,
+    key::{AlgorithmId, SlotId},
+    serialization::*,
+    transaction::Transaction,
+    yubikey::YubiKey,
+    Buffer,
 };
 use log::error;
+use rsa::{PublicKey, RSAPublicKey};
+use x509_parser::{parse_x509_der, x509::SubjectPublicKeyInfo};
 use zeroize::Zeroizing;
 
+/// An encoded point for some elliptic curve.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum EcPoint {
+    /// A point in compressed form.
+    Compressed {
+        /// EC algorithm
+        algorithm: AlgorithmId,
+
+        /// Encoded point
+        bytes: Buffer,
+    },
+
+    /// A point in uncompressed form.
+    Uncompressed {
+        /// EC algorithm
+        algorithm: AlgorithmId,
+
+        /// Encoded point
+        bytes: Buffer,
+    },
+}
+
+/// Information about a public key within a [`Certificate`].
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum PublicKeyInfo {
+    /// RSA keys
+    Rsa {
+        /// RSA algorithm
+        algorithm: AlgorithmId,
+
+        /// Public key
+        pubkey: RSAPublicKey,
+    },
+
+    /// EC keys
+    Ec(EcPoint),
+}
+
+impl PublicKeyInfo {
+    fn parse(subject_pki: &SubjectPublicKeyInfo<'_>) -> Result<Self, Error> {
+        match subject_pki.algorithm.algorithm.to_string().as_str() {
+            // RSA encryption
+            "1.2.840.113549.1.1.1" => {
```

Comment:
No better way to represent OIDs than this? I mean, it's fine and all, but...

Reply:
Best we can do for now. I've extracted these strings as constants.

Reply:
Opened https://github.com/rusticata/der-parser/issues/23